crates/web-plugins/didcomm-messaging/protocols/forward/src/web/handler.rs

```diff
@@ -1,14 +1,14 @@
-use super::routing::handler;
-use axum::response::Response;
+use crate::{web::routing::handler, ForwardError};
 use didcomm::Message;
 use shared::state::AppState;
+use std::sync::Arc;
 
 /// Mediator receives forwarded messages, extract the next field in the message body, and the attachments in the message
 /// then stores the attachment with the next field as key for pickup
 pub async fn mediator_forward_process(
-    state: &AppState,
+    state: Arc<AppState>,
     payload: Message,
-) -> Result<Message, Response> {
-    let result = handler(state, payload).await.unwrap();
+) -> Result<Option<Message>, ForwardError> {
+    let result = handler(state.clone(), payload).await.unwrap();
```

Comment:
Is this `clone()` necessary?

Reply:
Yes. Otherwise, the `Arc` value would have been moved and we would no longer have been able to access it in the `process_response` function.